kymatio/scattering2d/frontend/torch_frontend.py

```diff
@@ -0,0 +1,189 @@
+            Must be either equal to `1` or `2`.
+        pre_pad : boolean
+            Controls the padding: if set to False, a symmetric padding is applied
+            on the signal. If set to true, the software will assume the signal was
+            padded externally.
+        Psi : dictionary
+            Contains the wavelets filters at all resolutions. See
+            filter_bank.filter_bank for an exact description.
+        Phi : dictionary
+            Contains the low-pass filters at all resolutions. See
+            filter_bank.filter_bank for an exact description.
+        M_padded, N_padded : int
+             Spatial support of the padded input.
+
+        Notes
+        -----
+        The design of the filters is optimized for the value L = 8.
+
+        pre_pad is particularly useful when cropping bigger images because
+        this does not introduce border effects inherent to padding.
+
+        """
+    def __init__(self, J, shape, L=8, max_order=2, pre_pad=False, backend=None):
+        super(Scattering2D_torch, self).__init__(J, shape, max_order=max_order)
+        self.pre_pad, self.L, self.backend = pre_pad, L, backend
+        self.build()
+
+    def build(self):
+        self.M, self.N = self.shape
+        # use the default backend if no backend is provided
+        if not self.backend:
+            from ..backend import torch_backend as backend
+            self.backend = backend
+        elif self.backend.name[0:5] != 'torch':
+            raise RuntimeError('This backend is not supported.')
+
+        if 2 ** self.J > self.shape[0] or 2 ** self.J > self.shape[1]:
+            raise RuntimeError('The smallest dimension should be larger than 2^J.')
+        self.M_padded, self.N_padded = compute_padding(self.M, self.N, self.J)
+        # pads equally on a given side if the amount of padding to add is an even number of pixels, otherwise it adds an extra pixel
+        self.pad = self.backend.Pad([(self.M_padded - self.M) // 2, (self.M_padded - self.M+1) // 2, (self.N_padded - self.N) // 2,
+                                (self.N_padded - self.N + 1) // 2], [self.M, self.N], pre_pad=self.pre_pad)
+        self.unpad = self.backend.unpad
+        self.create_and_register_filters()
+
+    def create_and_register_filters(self):
+        """ This function run the filterbank function that
+            will create the filters as numpy array, and then, it
+            saves those arrays as module's buffers."""
+
+        # Create the filters
+        filters = filter_bank(self.M_padded, self.N_padded, self.J, self.L)
+        n = 0
+        self.phi, self.psi = filters['phi'], filters['psi']
+        for c, phi in self.phi.items():
+            if isinstance(c, int):
+                self.phi[c] = torch.from_numpy(self.phi[c]).unsqueeze(-1) # add a trailing singleton dimension to mark
+                # it as non-complex
+                self.register_buffer('tensor' + str(n), self.phi[c])
```

Comment:
do you get a correspondence between buffer numbers and filters? Otherwise consider using better names ...

Reply:
cf below